src/main/java/reposense/git/GitClone.java

```diff
@@ -55,7 +55,7 @@ public static CommandRunnerProcess cloneShallowBareAsync(RepoConfiguration confi
             String outputFolderName, LocalDateTime sinceDate) throws GitCloneException {
         try {
             return CommandRunner.runCommandAsync(rootPath,
-                                                    getCloneShallowBareCommand(config, outputFolderName, sinceDate));
+                    getCloneShallowBareCommand(config, outputFolderName, sinceDate));
```

Comment:
The tests for GitClone seems lacking compared to the other Git classes. Perhaps we could figure out a way to test the various clone variations.

Reply:
In a strange coincidence, I did voice the same thought some time back and the response is here https://github.com/reposense/RepoSense/pull/1562#issuecomment-877829751 under point 2.

I think in general, cloning is quite expensive so they can be done together with the system tests. If the system tests work, then the cloning likely works with it. There is an argument to be made for `cloneFromBareAndUpdateBranch` which is a local-to-local clone done while analyzing but I'm not sure how we would go about testing it.

Reply:
I think it makes to test these with the system tests, since the issues seem to be with system-specific quirks (although I'm not sure how much of GitClone is covered by the systemtests). I think the linked comment is probably right about tests in that we can only go to the extent of ensuring that the methods return the correct Strings and throw the right exceptions.